test/alu_8bit_sbc_test.c

```diff
@@ -0,0 +1,248 @@
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, h, &cpu.HL.bytes.high);
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 1));
+}
+
+Test(sbc_a_n, sbc_a_l_res_only, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x9D;
+    const uint8_t a = rand() % (0xFF);
+    const uint8_t l = rand() % (0xFF);
+    const uint8_t expected = a - l - 1;
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, l, &cpu.HL.bytes.low);
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 1));
+}
+
+Test(sbc_a_n, sbc_a_hl_res_only, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x9E;
+    const uint8_t a = 70;
+    const uint8_t num = 1;
+    const uint8_t expected = a - num - 1;
+    set_flag(1, C_FLAG);
+
+    const uint16_t address = rand() % (0xFFFF);
+    const uint16_t word_address = (rand() + 0x8000) % (0x10000);
+
+    // setup cpu
+    cpu.PC = address;
+    cpu.AF.bytes.high = a;
+    cpu.HL.word = word_address;
+    mmu_write_byte(address, opcode);
+    mmu_write_byte(word_address, num);
+
+    // do the actual emulation
+    cpu_step();
+    const uint8_t actual = cpu.AF.bytes.high;
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 1));
+}
+
+Test(sbc_a_n, sbc_a_d8_res_only, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0xDE;
+    const uint8_t a = 70;
+    const uint8_t num = 1;
+    const uint8_t expected = a - num - 1;
+    set_flag(1, C_FLAG);
+
+    const uint16_t address = rand() % (0xFFFF - 1);
+
+    // setup cpu
+    cpu.PC = address;
+    cpu.AF.bytes.high = a;
+    mmu_write_byte(address, opcode);
+    mmu_write_byte(address+1, num);
+
+    // do the actual emulation
+    cpu_step();
+    const uint8_t actual = cpu.AF.bytes.high;
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 2));
+}
+
+Test(sbc_a_n, sbc_check_integer_promotion, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x9D; // not important that it's exactly this one
+    const uint8_t a = 123;
+    const uint8_t l = 255;
+    const uint8_t expected = (uint8_t) (a - l - 1);
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, l, &cpu.HL.bytes.low);
+
+    // C should be set because of borrow
+    // checks if integer promotion was successful
+    cr_expect(eq(u8, get_flag_bit(C_FLAG), 1));
+
```

Comment:
Flag checks for other flags than C

Reply:
As the other flags have been tested in the first three unit tests, this is not the goal of this unit test.
Here we simply want to check for the integer promotion, which only includes the C flag in this specific edge case